### DISCUSSION TITLE
Forge API and Flux

### DISCUSSION THREAD

**Discussion by osi1880vr** [lllyasviel/stable-diffusion-webui-forge#2258] [Q&A]
Hi There,

I'm kind of a heavy API user of Forge since looong time. I did not update for quite some time due to the heavy changes that have been made, until now where I found the API is up and running again and everything is good again with the new (old) Forge. Well done, respect for all your hard work =)

Here comes my question: by when do we expect the API to allow to send in the Fulx relevant parameters during txt2img? By now I can run flux models via the API but they come out broken most the time maybe due to wrong VAE or using the wrong clip/t5 or something  I just dont see its missing.

Is there any date available? Or is it already possible and I did not see it in the FastAPI documentation?

If you wonder why I like the API so much, have a look at my baby https://github.com/osi1880vr/prompt_quill

Thanks for your time

**Comment by catto123455** [comment]
i am waiting too，i use this，and returned image is pure black. ：
import requests
import base64
from PIL import Image
from io import BytesIO
import os
import json  # 导入 json 模块

# 使用的API接口基础URL
web_ui_url = "http://xxxxxx:7862"  # 替换为你的服务地址和端口

# 构造 payload：文本到图像需要的参数
payload = {
    "prompt": "1girl",  # 正向提示词
    "negative_prompt": "low quality, bad anatomy, text",  # 反向提示词
    "seed": -1,  # 随机种子（-1 表示随机）
    "steps": 20,  # 采样步数
    "cfg_scale": 1,  # CFG Scale
    "width": 896,  # 图像宽度
    "height": 1152,  # 图像高度
    "batch_size": 1,  # 每批图像数量
    "n_iter": 1,  # 生成批次
    "sampler_index": "Euler",  # 采样方法
    "schedule_type": "Simple",  # 调度类型
    "send_images": True,  # 是否返回图片数据
    "save_images": False,  # 是否在服务端保存图像
    "override_settings": {
        "sd_model_checkpoint": "flux-dev-fp8.safetensors",  # 模型名称
        "sd_vae": "flux_ae.safetensors",  # VAE 名称
    }
}

# 向Stable Diffusion WebUI API发送Post请求以生成图片
response = requests.post(f"{web_ui_url}/sdapi/v1/txt2img", json=payload)

# 检查响应状态，成功时处理
if response.status_code == 200:
    json_data = response.json()  # 读取返回的JSON数据
    images = json_data['images']  # 获取生成的图像Base64编码数组

    # 指定保存路径
    output_dir = r"E:\output"  # 保存图片的目录
    os.makedirs(output_dir, exist_ok=True)  # 如果目录不存在，则创建目录

    # 处理每张图片，将Base64图像解码，然后使用PIL保存/显示
    for i, img_data in enumerate(images):
        img_bytes = base64.b64decode(img_data)  # 将Base64数据解码为字节流
        img = Image.open(BytesIO(img_bytes))  # 将字节流转换为PIL图像对象

        # 保存图片到指定目录
        output_path = os.path.join(output_dir, f"output_image_{i}.png")
        img.save(output_path)  # 保存图片
        print(f"图片已保存到: {output_path}")

        # 保存生成参数到文本文件
        parameters = json_data.get('parameters', {})  # 获取参数，默认为空字典
        with open(os.path.join(output_dir, f"output_image_{i}_parameters.txt"), "w", encoding="utf-8") as f:
            f.write(json.dumps(parameters, indent=4))  # 将字典转换为 JSON 字符串并写入文件

        img.show()  # 显示图片

else:
    print(f"Error: {response.status_code}, {response.text}")

**Reply by jtabernik** [reply]
I was having the same issue.  No idea if this will work for you or not, but I added these two parameters to my payload and it fixed it.  I see you have scheduler and sampler specified already, but with different parameter names.  Anyway, hope this helps!

```
    "sampler_name": "Euler",
    "scheduler": "Simple"
```

**Reply by catto123455** [reply]
i changed scripts，now it could run.
import requests
import base64
from PIL import Image
from io import BytesIO
import os
import json
from datetime import datetime

# 使用的API接口基础URL
web_ui_url = "http://xxx.xx.xxx.xx:7862"  # 替换为你的服务地址和端口

# 指定文本文件路径
txt_file_path = r"E:\工作日志\hbs6000.txt"

# 指定保存路径
output_dir = r"E:\output"  # 保存图片的目录
os.makedirs(output_dir, exist_ok=True)  # 如果目录不存在，则创建目录

# 读取文本文件
with open(txt_file_path, "r", encoding="utf-8") as file:
    lines = file.readlines()  # 读取所有行

# 遍历每一行
for line_number, line in enumerate(lines):
    # 去掉行末的换行符
    line = line.strip()

    # 如果行为空，跳过
    if not line:
        continue

    # 构造 prompt，追加 "solo,simple background,<lora:game_assets_v3:1>"
    prompt = f"{line}, solo, simple background, <lora:game_assets_v3:1>"

    # 构造 payload：文本到图像需要的参数
    payload = {
        "batch_size": 1,
        "cfg_scale": 1,
        "width": 512,
        "height": 512,
        "n_iter": 1,
        "negative_prompt": "",
        "override_settings": {
            "sd_model_checkpoint": "flux1-dev-bnb-nf4-v2",  # 模型名称
        },
        "prompt": prompt,  # 使用当前行的内容作为 prompt
        "restore_faces": False,
        "sampler_name": "[Forge] Flux Realistic",
        "scheduler": "Simple",
        "seed": -1,
        "steps": 20,
        "tiling": False
    }

    # 向Stable Diffusion WebUI API发送Post请求以生成图片
    response = requests.post(f"{web_ui_url}/sdapi/v1/txt2img", json=payload)

    # 检查响应状态，成功时处理
    if response.status_code == 200:
        json_data = response.json()  # 读取返回的JSON数据
        images = json_data['images']  # 获取生成的图像Base64编码数组

        # 获取当前时间戳
        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")  # 格式：年月日_时分秒

        # 处理每张图片，将Base64图像解码，然后使用PIL保存/显示
        for i, img_data in enumerate(images):
            img_bytes = base64.b64decode(img_data)  # 将Base64数据解码为字节流
            img = Image.open(BytesIO(img_bytes))  # 将字节流转换为PIL图像对象

            # 生成唯一的文件名（带时间戳和行号）
            output_path = os.path.join(output_dir, f"output_image_line{line_number + 1}_{timestamp}_{i}.png")

            # 保存图片到指定目录
            img.save(output_path)  # 保存图片
            print(f"图片已保存到: {output_path}")

            # 保存生成参数到文本文件
            parameters = json_data.get('parameters', {})  # 获取参数，默认为空字典
            param_file_path = os.path.join(output_dir, f"output_image_line{line_number + 1}_{timestamp}_{i}_parameters.txt")
            with open(param_file_path, "w", encoding="utf-8") as f:
                f.write(json.dumps(parameters, indent=4))  # 将字典转换为 JSON 字符串并写入文件

            img.show()  # 显示图片

    else:
        print(f"Error: {response.status_code}, {response.text}")

**Reply by destepar** [reply]
Hello guys,

Sorry to bother. Quite New in API world using Forge in Pinokio, I would like to have the endpoints working for you. Because /sdapi/v1/txt2img is not even in Docs.

Could you share with me how do you request to the API?
Thanks!

**Reply by Sholkof** [reply]
@POST("/sdapi/v1/txt2img") work for my app. Be sure to use scheduler: String = "Simple" in Request for Flux; --api --listen in set COMMANDLINE_ARGS; Check IP server Forge; open port 7860 in Firewall. I don't know how Forge works in Pinokio.
